tests/test_simple_target_loop.py

```diff
@@ -0,0 +1,188 @@
+
+        # send enable; new state is ENABLED
+        state_coro = self.scheduler_remote.evt_summaryState.next(timeout=1.)
+        cmd_attr = getattr(self.scheduler_remote, f"cmd_enable")
+        await cmd_attr.start(cmd_attr.DataType(), timeout=1.)
+
+        self.assertEqual(self.scheduler.summary_state, salobj.State.ENABLED)
+
+        state = await state_coro
+        self.assertEqual(state.summaryState, salobj.State.ENABLED,
+                         "Scheduler in %s, expected ENABLED. " % salobj.State(state.summaryState))
+
+    async def enable_queue(self):
+        """ Utility method to enable the Queue. """
+        # send enable; new state is ENABLED
+        cmd_attr = getattr(self.queue_remote, f"cmd_enable")
+        await cmd_attr.start(cmd_attr.DataType(), timeout=1.)
+
+        self.assertEqual(self.queue.summary_state, salobj.State.ENABLED)
+
+    def test_simple_loop(self):
+        """Test the simple target production loop."""
+
+        async def doit():
+
+            # Test 1 - Enable scheduler Queue is not enable. Scheduler should go to ENABLE and then to FAULT
+            # It may take some time for the scheduler to go to FAULT state.
+
+            state_coro = self.scheduler_remote.evt_summaryState.next(timeout=1.)
+            await self.enable_scheduler()
+
+            self.assertEqual(self.scheduler.summary_state, salobj.State.ENABLED,
+                             "Scheduler in %s, expected ENABLED. " % salobj.State(self.scheduler.summary_state))
+            # I won't check for self.scheduler.summary_state here because the scheduler can transition to Fault
+            # in the meantime
+
+            state = await state_coro
+            self.assertEqual(state.summaryState, salobj.State.FAULT,
+                             "Scheduler in %s, expected FAULT. " % salobj.State(state.summaryState))
+            self.assertEqual(self.scheduler.summary_state, salobj.State.FAULT,
+                             "Scheduler in %s, expected FAULT. " % salobj.State(state.summaryState))
+
+            # recover from fault state sending it to OFFLINE
+            cmd_attr = getattr(self.scheduler_remote, f"cmd_exitControl")
+            await cmd_attr.start(cmd_attr.DataType(), timeout=1.)
+
+            self.assertEqual(self.scheduler.summary_state, salobj.State.OFFLINE,
+                             "Scheduler in %s, expected OFFLINE. " % salobj.State(self.scheduler.summary_state))
+
+            # now enable queue...
+            await self.enable_queue()
+
+            # ...and try again. This time the scheduler should stay in enable and publish targets to the queue.
+
+            def assertEnable(data):
+                """Callback function to make sure scheduler is enabled"""
+                self.assertEqual(data.summaryState, salobj.State.ENABLED,
+                                 "Scheduler unexpectedly transitioned from "
+                                 "ENABLE to %s" % salobj.State(data.summaryState))
+
```

Comment:
Consider breaking this up into two tests, to make it easier to follow what is going on:
• One tests that enable eventually ends up in FAULT and that you can then fix that.
• The other test starts out with the queue enabled and you test that it the scheduler goes to ENABLED and stays there while running.